app/jobs/appointment_notification/worker.rb

```diff
@@ -6,50 +6,77 @@ class AppointmentNotification::Worker
 
   DEFAULT_LOCALE = :en
 
-  def perform(appointment_id, communication_type, locale = nil)
-    appointment = Appointment.find_by(id: appointment_id)
-    unless appointment
-      logger.warn "Appointment #{appointment_id} not found, skipping notification"
+  def perform(appointment_reminder_id, communication_type)
+    reminder = AppointmentReminder.includes(:appointment, :patient).find(appointment_reminder_id)
+    return if reminder.appointment.previously_communicated_via?(communication_type)
```

Comment:
This will have to change for the experiment because we can send multiple of the same type. I have a version of that logic on another branch. But I think this should work for non-experiment reminders.